### ISSUE TITLE
Parents of NgModel are not reset when using custom value accessors

### ISSUE THREAD

**Issue by smasala** [angular/angular#17294]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Take following multiple nested reusable components `custom-cmp` and `nested-cmp`:

```
<custom-cmp [(ngModel)]="bob">
    <nested-cmp [(ngModel)]="bob" *ngIf="something">
        <input [(ngModel)]="bob" />
    </nested-cmp>
    .... further cmps ....
</custom-cmp>
```
Each reusable component has a custom value accessor defined to pass the `[(ngModel)]` down to the `input` element.

When the input element is changed, through `keystroke` for example... all components: `custom-cmp`, `nested-cmp` and `input` receive the correct `ng-dirty` flags. The `NgModel` at all three levels are also `dirty: true`.

When `reset()` is called on the `input` or `nested-cmp` model (DI: `NgModel` or via the `@ViewChild(NgModel)` decorator) for example, none of the parents are notified and reset, as the `parent` property of the model in question is suddenly `undefined`.

For example `nested-cmp` component class:

```
@ViewChild(NgModel) myInputModel: NgModel // this is the ngModel of the 'input' element (a child of nested-cmp)

doReset(): void {
    this.myInputModel.reset(...original value...);
}
```

In the above example: `doReset()` - only the `input element ngModel` is reset... i.e the input is set back to `ng-pristine`. The host, `nested-cmp` and it's parent `custom-cmp` remains dirty (`ng-dirty`). 

**Expected behavior**
That all parent components on which `ngModel.reset()` is called are restored to their `pristine` states.

**Minimal reproduction of the problem with instructions**
Full reproduction:
http://plnkr.co/edit/TTPfmhRj1LN2ZXjz4JXo?p=preview

Further comments in same thread:
https://github.com/angular/angular/issues/16772#issuecomment-306779757
https://github.com/angular/angular/issues/16772#issuecomment-306784580
https://github.com/angular/angular/issues/16772#issuecomment-306784810

**What is the motivation / use case for changing the behavior?**
Creating custom components and taking advantage of two-way-binding without forms

**Please tell us about your environment:**

* **Angular version:** 2.4.9

* **Browser:** Chrome 58
 
* **Language:** TypeScript 2.2.2

**Comment by ericsnap** [comment]
Came across this thread as I was looking for a solution to the reverse problem. In my case, I want to _reset_ the parent form control and have it propagate to its children.
My child controls are all implementing `ControlValueAccessor`. Am I understanding correctly that when `writeValue()` is called, is the right place to reset my child component? when `writeValue()` is called, I should consider my child control _pristine_ and _untouched_?

**Comment by mlc-mlapis** [comment]
@ericsnap ... not sure what you want to do but `writeValue()` processes the data update on the direction from a model to a view.

**Comment by ericsnap** [comment]
@mlc-mlapis right, I think that's what I gathered from the documentation. Conceptually, if data gets updated on the model (the outside model in our case since we are dealing with child components), are we correct in assuming we should consider our child component pristine and untouched at this point (until the user modified the view again, of course)?

**Comment by mlc-mlapis** [comment]
@ericsnap ... it depends on a case ... because in the situation when a live edit is available from more sides (not only from one user in the view) ... imagine using some push technologies (WebSocket) then even the change through `writeValue()` is a user change ... from another side.

**Comment by smasala** [comment]
Angular 7.2.x can still inject `NgControl`
